diff --git a/up.go b/up.go
index bf0315860..4c01ce6a8 100644
--- a/up.go
+++ b/up.go
@@ -21,13 +21,13 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/BARGHEST-ngo/MESH/analyst/internal/client/tailscale"
 	shellquote "github.com/kballard/go-shellquote"
 	"github.com/peterbourgon/ff/v3/ffcli"
 	"tailscale.com/feature/buildfeatures"
 	_ "tailscale.com/feature/condregister/identityfederation"
 	_ "tailscale.com/feature/condregister/oauthkey"
 	"tailscale.com/health/healthmsg"
-	"tailscale.com/internal/client/tailscale"
 	"tailscale.com/ipn"
 	"tailscale.com/ipn/ipnstate"
 	"tailscale.com/net/netutil"
