diff --git a/cmd/tailscale/cli/cli.go b/cmd/tailscale/cli/cli.go
index 5ebc23a5b..862c1b1e6 100644
--- a/cmd/tailscale/cli/cli.go
+++ b/cmd/tailscale/cli/cli.go
@@ -20,6 +20,7 @@
 	"text/tabwriter"
 	"time"
 
+	"github.com/i582/cfmt/cmd/cfmt"
 	"github.com/mattn/go-colorable"
 	"github.com/mattn/go-isatty"
 	"github.com/peterbourgon/ff/v3/ffcli"
@@ -120,6 +121,12 @@ func Run(args []string) (err error) {
 		})
 	})
 
+	cfmt.Print(`
+	{{[ 𝗠𝗘𝗦𝗛 ]}}::blue
+	{{[ 𝐀𝐍𝐀𝐋𝐘𝐒𝐓 𝐂𝐋𝐈𝐄𝐍𝐓 ]}}::blue
+	{{by Barghest.asia. No rights reserved.}}::blue
+	`)
+
 	rootCmd := newRootCmd()
 	if err := rootCmd.Parse(args); err != nil {
 		if errors.Is(err, flag.ErrHelp) {
@@ -233,16 +240,19 @@ func newRootCmd() *ffcli.Command {
 
 	var rootCmd *ffcli.Command
 	rootCmd = &ffcli.Command{
-		Name:       "tailscale",
-		ShortUsage: "tailscale [flags] <subcommand> [command flags]",
-		ShortHelp:  "The easiest, most secure way to use WireGuard.",
+		Name:       "meshcli",
+		ShortUsage: "meshcli [flags] <subcommand> [command flags]",
+		ShortHelp:  "Censorship-resisting overlay network for digital forensics.",
 		LongHelp: strings.TrimSpace(`
-For help on subcommands, add --help after: "tailscale status --help".
+For help on subcommands, add --help after: "meshcli status --help".
 
 This CLI is still under active development. Commands and flags will
 change in the future.
 `),
 		Subcommands: nonNilCmds(
+			adbpairCmd,
+			adbcollectCmd,
+			adbdisableCmd,
 			upCmd,
 			downCmd,
 			setCmd,
@@ -254,7 +264,7 @@ func newRootCmd() *ffcli.Command {
 			netcheckCmd,
 			ipCmd,
 			dnsCmd,
-			statusCmd,
+			meshStatusCmd(),
 			metricsCmd,
 			pingCmd,
 			ncCmd,
