diff --git a/cmd/tailscaled/tailscaled.go b/cmd/tailscaled/tailscaled.go
index 410ae00bc..189b944c7 100644
--- a/cmd/tailscaled/tailscaled.go
+++ b/cmd/tailscaled/tailscaled.go
@@ -27,6 +27,7 @@
 	"syscall"
 	"time"
 
+	"tailscale.com/cmd/tailscale/cli"
 	"tailscale.com/cmd/tailscaled/childproc"
 	"tailscale.com/control/controlclient"
 	"tailscale.com/envknob"
@@ -134,6 +135,7 @@ func defaultPort() uint16 {
 	installSystemDaemon   func([]string) error                      // non-nil on some platforms
 	uninstallSystemDaemon func([]string) error                      // non-nil on some platforms
 	createBIRDClient      func(string) (wgengine.BIRDClient, error) // non-nil on some platforms
+	cliRun                func([]string) error                      = cli.Run
 )
 
 // Note - we use function pointers for subcommands so that subcommands like
@@ -144,6 +146,7 @@ func defaultPort() uint16 {
 	"install-system-daemon":   &installSystemDaemon,
 	"uninstall-system-daemon": &uninstallSystemDaemon,
 	"be-child":                &beChildFunc,
+	"cli":                     &cliRun,
 }
 
 var beCLI func() // non-nil if CLI is linked in with the "ts_include_cli" build tag
